back/lib/tasks/single_use/20250728_reprocess_citizen_images.rake

```diff
@@ -0,0 +1,131 @@
+namespace :single_use do
+  # Reprocesses citizen images, to ensure the relatively new metadata stripping is
+  # applied. Primarily for use with the Vienna tenant, but tenant is selectable
+  # to enable testing on a demo.
+  # Reprocesses user avatars, idea_images, and Idea text_images.
+  # Optionally, a date can be specified to only reprocess images updated before that date.
+  # If no date is specified, it defaults to tomorrow's date, effectively giving no cutoff.
+  # Usage: `rake single_use:reprocess_citizen_images[host,date]`
+  # Example: `rake single_use:reprocess_citizen_images[simonssandbox.govocal.com,'2023-07-01']`
+  desc 'Reprocesses ALL CarrierWave versions and the original image for specific models and mounted uploader.'
+  task :reprocess_citizen_images, %i[host date] => :environment do |_t, args|
+    # Reduce logging when developing (to more closely match the production environment)
+    # dev_null = Logger.new('/dev/null')
+    # Rails.logger = dev_null
+    # ActiveRecord::Base.logger = dev_null
+
+    tenant_host = args[:host]
+    cutoff_date = args[:date] ? Date.parse(args[:date]) : Time.zone.today + 1.day
+
+    puts "date: #{cutoff_date.inspect}"
+
+    puts "Overriding `#{BaseImageUploader.name}#filename` to prevent renaming during reprocessing..."
+
+    BaseImageUploader.class_eval do
+      define_method(:filename) do
+        current_db_value = model.read_attribute(mounted_as)
+
+        if current_db_value.present?
+          File.basename(current_db_value.to_s)
+        end
+      end
+    end
+
+    def reprocess_images(records, uploader_mount, model_name, tenant_host)
+      total_reprocessed = 0
+      errored_records = []
+
+      records.each do |record|
+        uploader = record.send(uploader_mount)
+
+        if uploader.present? && uploader.file.present?
+          puts "  Reprocessing image for #{model_name} ID: #{record.id}..."
+
+          uploader.cache_stored_file!
+          uploader.retrieve_from_cache!(uploader.cache_name)
+          uploader.recreate_versions!
```

Comment:
Do you confirm it also reprocess the original file?

Reply:
Yes. I have checked the original file has (significantly) less metadata tags after reprocessing (using exiftool).